src/EditorFeatures/Core/EditorConfigSettings/Updater/SettingsUpdateHelper.cs

```diff
@@ -83,17 +88,16 @@ internal static partial class SettingsUpdateHelper
                         { Severity: ReportDiagnostic.Error } => "error",
                         _ => string.Empty
                     };
-                    optionValue = $"{optionValue}:{severity}";
+                    return (true, optionName, optionValue, diagnosticIds, severity, language);
                 }
 
-                var language = option.IsPerLanguage ? Language.CSharp | Language.VisualBasic : Language.CSharp;
-                return (true, optionName, optionValue, language);
+                return (true, optionName, optionValue, ImmutableArray<string>.Empty, null, language);
             }
         }
 
         public static SourceText? TryUpdateAnalyzerConfigDocument(SourceText originalText,
                                                                   string filePath,
-                                                                  IReadOnlyList<(string option, string value, Language language)> settingsToUpdate)
+                                                                  IReadOnlyList<(string option, string value, ImmutableArray<string> ids, string? severity, Language language)> settingsToUpdate)
```

Comment:
so we don't really use this alignment of parametres in C#.  it's hard to maintain (i.e. when method name changes) and it takes up a ton of horizontal whitespace.  cosider just indenting 4 from teh method level.